ert_shared/ensemble_evaluator/entity/ensemble.py

```diff
@@ -125,10 +143,33 @@ def build(self):
         return _ScriptJob(self._id, self._name, self._executable, self._args)
 
 
+class _FunctionJobBuilder(_BaseJobBuilder):
+    def __init__(self):
+        super().__init__()
+        self._fun = None
+        self.reset()
+
+    def reset(self):
+        super().reset()
+        self._fun = None
+        return self
```

Comment:
Could I ask the reason for returning self in these functions?

Reply:
It allows for a builder pattern, so you can do:

`create_ensemble_builder(...).add_job(...).do_some_stuff(...).reset(...).do_some_stuff(..).build(...)`

Reply:
https://refactoring.guru/design-patterns/builder